doc/md/versioned-migrations.md

```diff
@@ -241,3 +241,115 @@ func main() {
    }
 }
 ```
+
+## Atlas migration directory integrity file
+
+Suppose you have multiple teams develop a feature in parallel and both of them need a migration. If Team A and Team B do
+not check in with each other, they might end up with a broken set of migration files (like adding the same column twice)
+since new files do not raise a merge conflict in a version control system like git. To prevent concurrent creation of
+new migration files and guard against accidental changes in the migration history, Atlas has a feature called __
+Migration Directory Integrity File__, which simply is another file in your migration directory called `atlas.sum`. 
+
+The `atlas.sum` file contains the checksum of each migration file (implemented by a reverse, one branch merkle hash
+tree), and a sum of all files. Adding new files results in a change to the sum file, which will raise merge conflicts in
+most version controls systems. This is an example of an `atlas.sum` file:
+
+```text
+h1:UtZn3IzU66t05/t1sybK48xAeeEvguhtMuhqsbIGMhs=
+1_initial.sql h1:81KZdrTEWJQ5UuzuOyU8C6R7rcXFOuHfVA4AAhS5OoU=
+```
+
+You can enable this feature by using the `schema.WithSumFile()` option.
+
+```go
+package main
+
+import (
+   "context"
+   "log"
+   "strings"
+   "text/template"
+   "time"
+
+   "ariga.io/atlas/sql/migrate"
+   "ariga.io/atlas/sql/sqltool"
+   "entgo.io/ent/dialect/sql"
+   "entgo.io/ent/dialect/sql/schema"
+   "entgo.io/ent/entc"
+   "entgo.io/ent/entc/gen"
+)
+
+func main() {
+   // Load the graph.
+   graph, err := entc.LoadGraph("/.schema", &gen.Config{})
```

Comment:
I'm curious why you don't use `<project>/ent/migrate.Tables` instead? It's defined statically, and also promises that the codegen already was executed for the schema.